engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetCullFaceMode.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag.stateChanges;
+
+import org.lwjgl.opengl.GL11;
+import static org.lwjgl.opengl.GL11.GL_BACK;
+import static org.lwjgl.opengl.GL11.GL_FRONT;
+import static org.lwjgl.opengl.GL11.GL_FRONT_AND_BACK;
+import org.terasology.rendering.dag.RenderPipelineTask;
+import org.terasology.rendering.dag.StateChange;
+import org.terasology.rendering.dag.tasks.SetCullFaceModeTask;
+
+/**
+ * TODO: Add javadocs
+ */
+public final class SetCullFaceMode implements StateChange {
+
+    private static SetCullFaceMode defaultInstance = new SetCullFaceMode(GL_BACK); // also specified in OpenGL documentation
+    private SetCullFaceModeTask task;
+    private int mode;
+
+    public SetCullFaceMode(int mode) {
+        this.mode = mode;
+        validate();
+    }
+
+    private void validate() {
```

Comment:
I see, but I think it's much cleaner even though it's just several lines. Of course, if you are strongly against it, I'll move them into the constructor. It's a tendency, mostly because of [this course ](http://aktemur.github.io/cs222/resources.html) and [Uncle Bob's famous book](https://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882).